src/reader/linearring.rs

```diff
@@ -118,3 +115,135 @@ impl<'a> LineStringTrait for &LinearRing<'a> {
         )
     }
 }
+
+impl<'a> GeometryTrait for LinearRing<'a> {
+    type T = f64;
+    type PointType<'b>
+        = Point<'a>
+    where
+        Self: 'b;
+    type LineStringType<'b>
+        = LinearRing<'a>
+    where
+        Self: 'b;
+    type PolygonType<'b>
+        = Polygon<'a>
```

Comment:
Does it make sense for LinearRing to have a PolygonType? I'd guess that it should be `Unimplemented` like you've done for the `writer::Rect` 

(Similar question for the rest of the associated types in this `impl GeometryTrait` and for the other `impl GeometryTrait`'s ...)

TBH I don't really know how it'd be mis-used, since we never return any of those types AFAICT, so maybe it doesn't really matter.

Reply:
I don't think it really matters.

I changed it in [`5eb5c1d` (#69)](https://github.com/georust/wkb/pull/69/commits/5eb5c1dc9b181343f5fb46a13c7cc0fced009450). Does that resolve your question?

Reply:
Agreed. Only the type(s) you're actually returning from `as_type` will ever be encountered.